cylc/uiserver/handlers.py

```diff
@@ -322,32 +315,37 @@ class UIServerGraphQLHandler(CylcAppHandler, TornadoGraphQLHandler):
     def set_default_headers(self) -> None:
         self.set_header('Server', '')
 
-    def initialize(self, schema=None, executor=None, middleware=None,
-                   root_value=None, graphiql=False, pretty=False,
-                   batch=False, backend=None, auth=None, **kwargs):
+    def initialize(
+        self, schema=None, middleware=None, root_value=None, pretty=False,
+        batch=False, execution_context_class=None, validation_rules=None,
+        auth=None, **kwargs
+    ):
         super(TornadoGraphQLHandler, self).initialize()
-        self.auth = auth
-        self.schema = schema
 
+        self.schema = schema or self.schema
+        self.auth = auth
         if middleware is not None:
-            self.middleware = list(self.instantiate_middleware(middleware))
+            if isinstance(middleware, MiddlewareManager):
+                self.middleware = middleware
+            else:
+                self.middleware = list(instantiate_middleware(middleware))
+        self.root_value = root_value
+        self.pretty = pretty or self.pretty
+        self.batch = batch or self.batch
+        self.execution_context_class = (
+            execution_context_class or self.execution_context_class
+        )
+
```

Comment:
With the exception of the `self.auth = auth` line, this duplicates `TornadoGraphQLHandler.initialize` so is overwriting it?!

I think we can replace this with:

```suggestion
        super(TornadoGraphQLHandler, self).initialize(
            schema=None,
            middleware=None,
            root_value=None,
            pretty=False,
            batch=False,
            execution_context_class=None,
            validation_rules=None,
        )
        self.auth = auth
```